### PR TITLE
AGDIGGER-42 - general clean-up in docker setup

### PR DESCRIPTION
**Pull request by wtrocki**
# Motivation

Cleanup configuration to include:

- Local build for the s2i image for image modifications
- Change directory to split between slave and jenkins image.

Note: image already pushed to aerogear repo. 

# Verification steps

Try to execute local build steps
Publish image to dockerhub (to check if you have permission to do so)
Setup new PersistentVolume
Change DeploymentConfig to include new image

# Expected results

Android node should be available by default.

### CODE REVIEW

**Comment by wtrocki** [comment]
ping @luigizuccarelli

File: docker/jenkins1-centos/contrib/kube-slave-common.sh [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+# the '!' handles env variable indirection so we can resolve the nested variable
+# see: http://stackoverflow.com/a/14204692
+JNLP_HOST=${!T_HOST}
+T_PORT=${JNLP_SERVICE_NAME}_SERVICE_PORT
+JNLP_PORT=${!T_PORT}
+
+export JNLP_PORT=${JNLP_PORT:-50000}
+
+ANDROID_SLAVE=docker.io/aerogear/jenkins-android-slave
+MAVEN_SLAVE=registry.access.redhat.com/aerogear/jenkins-android-slave
```
Comment (luigizuccarelli) [inline docker/jenkins1-centos/contrib/kube-slave-common.sh]: @wtrocki - If we refer to redhat registry here - would this not need the user to have some form of subscription to pull this image

File: docker/jenkins1-centos/contrib/kube-slave-common.sh [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+# if the master is running the centos image, use the centos slave images.
+if [[ `grep CentOS /etc/redhat-release` ]]; then
+  ANDROID_SLAVE=docker.io/aerogear/jenkins-android-slave
+  MAVEN_SLAVE=openshift/jenkins-slave-maven-centos7
+fi
```
Comment (luigizuccarelli) [inline docker/jenkins1-centos/contrib/kube-slave-common.sh]: @wtrocki - You can ignore my previous comment - this addresses the question :+1:

File: docker/jenkins1-centos/contrib/kube-slave-common.sh [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+          <instanceCap>2147483647</instanceCap>
+          <label>maven</label>
+          <volumes/>
+          <envVars/>
+          <nodeSelector/>
+          <remoteFs>/tmp</remoteFs>
+          <serviceAccount>${oc_serviceaccount_name}</serviceAccount>
+        </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+        <org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
+          <name>android</name>
```
Comment (luigizuccarelli) [inline docker/jenkins1-centos/contrib/kube-slave-common.sh]: @wtrocki - As mentioned this is a hardcoded version to get the POC working. We will look at other ways to do this better

Comment (matzew) [inline docker/jenkins1-centos/contrib/kube-slave-common.sh]: So this defines `node (android)` in the pipeline, spins up an 'Android" container/slave ?

Comment (wtrocki) [inline docker/jenkins1-centos/contrib/kube-slave-common.sh]: Yes!

**Comment by matzew** [comment]
:+1:  ship it

**Pull request merged**